lib/inspec/control_eval_context.rb

```diff
@@ -53,6 +53,7 @@ def initialize(backend, conf, dependencies, require_loader)
           @dependencies = dependencies
           @require_loader = require_loader
           @skip_file = false
+          @skip_only_if_eval = conf['check_mode'] || false
```

Comment:
I'd like to draw a line at what things care about the concept of "check mode". In my opinion, the line needs to be drawn at the profile. Controls (and their evaluation contexts) should not care about the mode in which InSpec was executed but rather whether they should care about only_if or not.

Can we find a way to gracefully pass a bool for skip_only_if_eval from the profile to the control instead of passing check_mode here?

Reply:
What are your thought about just passing it all the way from the CLI, e.g. `o[:skip_only_if_eval] = true` vs `o[:check_mode]`?

Reply:
What are your thoughts about just passing it all the way from the CLI, e.g. `o[:skip_only_if_eval] = true` vs `o[:check_mode] = true`?

Reply:
While you can claim YAGNI, what if something else in the future needs to care about whether we're in check_mode or not? And should the CLI really be directing how a control is evaluated?

The CLI knows it's in check mode, the profile needs to know it's in check mode (because it's the thing being checked), and then the profile should use that information to determine whether or not certain activities should take place. I still believe the line I drew and described above is the right separation of duties and logic.

Reply:
I agree, I moved the line up to Profile and then passed `skip_only_if_eval` it to the `ControlEvalContext`